main.cpp

```diff
@@ -344,12 +269,30 @@ struct DoubleType
     DoubleType& multiply( double rhs );
     DoubleType& divide( double rhs );
 
-    operator double() { return *value; }
+    DoubleType& pow(double rhs);
+    DoubleType& pow(const FloatType& rhs);
+    DoubleType& pow(const DoubleType& rhs);
+    DoubleType& pow(const IntType& rhs);
+
+    operator double() const { return *value; }
 
 private:
     double* value;
+    DoubleType& powInternal(double arg);
 };
 
+DoubleType& DoubleType::pow(double rhs)
+{
+    powInternal(rhs);
+    return *this;
+}
```

Comment:
what does `powInternal` return?
return the result of calling `powInternal`
```
return powInternal()
```
- [ ] revise